### PR TITLE
vala: look for files in `targetOffset`

### PR DESCRIPTION
**Pull request by Artturin**
when vala is in `nativeBuildInputs`, look for files in `buildInputs`.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome/apps/gnome-clocks/default.nix
Comment (Artturin) [inline pkgs/desktops/gnome/apps/gnome-clocks/default.nix:1]: `pkgsCross.aarch64-multiplatform.gnome.gnome-clocks` doesn't build anymore :(

fails with

```
error: builder for '/nix/store/s617q8r4jjzbk9ggqkdaxm40qfrdgjl1-qtsvg-5.15.10.drv' failed with exit code 1;
       last 3 log lines:
       > Error: detected mismatched Qt dependencies:
       >     /nix/store/br4gj9l8s4zlf0w66zscvhznvg0q45zj-qtbase-5.15.10-dev
       >     /nix/store/jfsn6q4hqr75fram7kvqjh1d00lp7dsa-qtbase-5.15.10-dev
       For full logs, run 'nix log /nix/store/s617q8r4jjzbk9ggqkdaxm40qfrdgjl1-qtsvg-5.15.10.drv'.
error: 1 dependencies of derivation '/nix/store/dk2c0f4lghxsjj7p93dh86wrr50jm2jl-python3.11-PyQt5-5.15.9.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jlkbwc40kwv6d9imv8jd8c3r1dcygp9x-ffado-2.4.7.drv' failed to build
error: 1 dependencies of derivation '/nix/store/75ma0q1ksl6a2hhi6ficlji76rvb7qnz-pipewire-0.3.80.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8w0nikpm948g5fg13l626wcsh6bj5ch0-SDL2-2.28.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/366dkm8yshzfylr78ja6r0p4jl8cyvxc-openal-soft-1.23.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kzaq6jq6923z8a6ssqx3zrv8g0ffm9zr-gst-plugins-bad-1.22.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/14hc07w3g6viw4ib8wcswklpkc8cj7i5-gtk4-4.10.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/818xgfgjayq8jmf9ksgv4r9qywasm0hi-wrap-gapps-hook.drv' failed to build
error: 1 dependencies of derivation '/nix/store/k5p6gikvndp1pl66s23rgmpmw7gdawa3-gnome-clocks-aarch64-unknown-linux-gnu-44.0.drv' failed to build
```

Comment (Artturin) [reply pkgs/desktops/gnome/apps/gnome-clocks/default.nix:1]: seahorse builds successfully

File: pkgs/test/cc-wrapper/cxx-make-unique.cc
Comment (Artturin) [inline pkgs/test/cc-wrapper/cxx-make-unique.cc:1]: Accidentally committed, will remove

File: pkgs/desktops/gnome/apps/gnome-clocks/default.nix
Comment (uninsane) [reply pkgs/desktops/gnome/apps/gnome-clocks/default.nix:1]: i've seen this failure recently, without the vala patch. i notice in pipewire that the ffado component is conditional:
```nix
ffadoSupport ? stdenv.buildPlatform.canExecute stdenv.hostPlatform
```
yet here we are building ffado, presumably for pipewire, and getting the same type of error as back when this was unconditional. maybe https://github.com/NixOS/nixpkgs/pull/242782 has some hints.

File: pkgs/desktops/gnome/apps/gnome-clocks/default.nix [hunk 1]
```diff
@@ -45,9 +45,6 @@ stdenv.mkDerivation rec {
     desktop-file-utils
     libxml2
     gobject-introspection # for finding vapi files
```
Comment (uninsane) [inline pkgs/desktops/gnome/apps/gnome-clocks/default.nix:47]: presumably this gobject-introspection can be removed while you're here too, going by the comment.

File: pkgs/desktops/gnome/apps/gnome-clocks/default.nix
Comment (Artturin) [reply pkgs/desktops/gnome/apps/gnome-clocks/default.nix:1]: Possibly caused by https://github.com/NixOS/nixpkgs/pull/264903 or https://github.com/NixOS/nixpkgs/pull/264965

These https://github.com/NixOS/nixpkgs/blob/ba1a718f343b934a5285ca12853a2df41d72465b/pkgs/development/libraries/qt-5/5.15/default.nix#L366-L377 are really fragile

**Review by uninsane — APPROVED** [review]
diff looks fine. too many rebuilds for me to `nixpkgs-review`, but i cross compiled and deployed this to my aarch64 box and it's working as expected 👍️

**Pull request merged**